usecases/usecases.go

```diff
@@ -0,0 +1,93 @@
+package usecases
+
+type ProgrammingLanguage string
+type UserId string
+type SnippetId string
+type CommentId string
+type Token string
+
+type User struct {
+	Id       UserId // Unique identifier, persists through username changes
+	Username string // Visible username, can be changed
+}
+
+type Rating struct {
+	Likes    int
+	Dislikes int
+}
+
+type Comment struct {
```

Comment:
In case of multiple comments from a user, how are you going to present them, in what order? There's no timestamp or any other field to define this order, to say nothing of emoji :polish_cow:, potential replies to comments etc.